### PR TITLE
Moderation background process gives memory limit error

### LINKED ISSUE #3153: Moderation background process gives memory limit error
**Describe the bug**
When we have a large number of data in the system. moderation background process gives memory limit error and stops working.

**Expected behavior**
It should work smoothly without having any issues.

**Screenshots**
https://screencast-o-matic.com/watch/crX6XoVl9yZ

**Environment**
Test with DB: https://drive.google.com/file/d/1lDWRfKtOt0zUr8MDmweuyfxKKvLQIwql/view?usp=sharing

**Suggestions**
- Create an individual background process for each and every item in moderation.
- If possible create a paginated item when finding the related contents.
- If all are done then create a new background process for the moderation and write migration of already placed background process will work on a new background.
- Needs scope: How to handle if background process when not finished.

### PR DESCRIPTION
**Pull request by KartikSuthar**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #3153

### How to test the changes in this Pull Request:

* [x] Create an individual background process for each and every item in moderation.
* [x] Implemented a paginated item when finding the related contents.
* [x] Re-scheduled the background process until the queue is not empty and not schedules

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-core/bp-core-actions.php [hunk 1]
```diff
@@ -344,3 +344,18 @@ function bb_email_handle_cron_healthcheck() {
...
+ * @since BuddyBoss [BBVERSION]
+ */
+function bp_handle_cron_healthcheck() {
+	global $bp_background_updater;
+	if ( $bp_background_updater->is_updating() ) {
+		$bp_background_updater->schedule_event();
+	}
+}
+
+add_action( 'bp_init_background_updater', 'bp_handle_cron_healthcheck' );
```
Comment (chetansatasiya) [inline src/bp-core/bp-core-actions.php:361]:
```suggestion
function bb_handle_cron_healthcheck() {
	global $bp_background_updater;
	if ( $bp_background_updater->is_updating() ) {
		$bp_background_updater->schedule_event();
	}
}

add_action( 'bp_init_background_updater', 'bb_handle_cron_healthcheck' );
```

**Review by chetansatasiya — APPROVED** [review]
Good Work LGTM 🚀

**Comment by sumitbuddyboss** [comment]
Hello @KartikSuthar 

A related issue is resolved, all the test cases are passed, Please find below test cases. after these changes, we have confirmed that all the moderation-related things are working as expected, please check the below video.

**Video**
https://www.loom.com/share/0f08c69bbc534b2c98d1d1fbc65fc1aa

**Test case:**
https://docs.google.com/spreadsheets/d/1v7bvYc-n-VZYQu8CcAcc9jLmHFduAQmxz_0n8lhjw7o/edit#gid=527040952

Thank you @JetalDobariya for your help.

Ready for release 🚀 

Thank you.

**Comment by JetalDobariya** [comment]
Hello @KartikSuthar 

I did verify this PR in qa-approved branch, Every Moderation features has been working fine.
Category Backend : https://prnt.sc/2144zkb
Category Frontend : https://prnt.sc/21452xs
Settings Backend : https://prnt.sc/2145bh1
Reported Content : https://prnt.sc/2145h7s
Blocked Member : https://prnt.sc/2145liq
Emails : https://prnt.sc/2145nfh
https://prnt.sc/2145onq
wp_options : https://prnt.sc/217vlgt

Thanks!
CC : @sumitbuddyboss @sourabhmatolia

**Pull request merged**